### DISCUSSION TITLE
Need some help with the Magic Mirror.

### DISCUSSION THREAD

**Discussion by LarryXO** [PlummersSoftwareLLC/NightDriverStrip#142] [Q&A]
I couldn't get it to compile until I added a code block to faneffects.h, around the end of the if block:

#elif FAN_SIZE == 100
static const int FanPixelsVertical[FAN_SIZE] =
{
  0, 99, 1, 98, 2, 97, 3, 96, 4, 95, 5, 94, 6, 93, 7, 92, 8, 91, 9, 90, 10, 89, 11, 88, 12, 87, 13, 86, 14, 85, 15, 84, 16, 83, 17, 82, 18, 81, 19, 80, 20, 79, 21, 78, 22, 77, 23, 76, 24, 75, 25, 74, 26, 73, 27, 72, 28, 71, 29, 70, 30, 69, 31, 68, 32, 67, 33, 66, 34, 65, 35, 64, 36, 63, 37, 62, 38, 61, 39, 60, 40, 59, 41, 58, 42, 57, 43, 56, 44, 55, 45, 54, 46, 53, 47, 52, 48, 51, 49, 50
};

I followed the numbering convention for the earlier definitions, assuming it was an oversight. However, this causes regular crashes on the actual hardware, although it will compile and upload. Can someone at least point me in the right direction on this?

Thanks in advance.

**Comment by davepl** [comment]
At some point I changed that table to a function, so it worked on and size.  But now I can’t find the code!

As you can see, it’s just a repeating pattern.  Code to generate that pattern is pretty straightforward, so that’s what I’d do - write the function and get rid of the table

- Dave
